zenoh-android/RobotTeleop/app/src/main/java/com/example/zenohapp/ui/examples/TeleopFragment.kt

```diff
@@ -0,0 +1,476 @@
+import android.widget.SeekBar
+import android.widget.SeekBar.OnSeekBarChangeListener
+import android.widget.Toast
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
+import com.example.zenohapp.R
+import com.example.zenohapp.ZenohViewModel
+import com.example.zenohapp.cdr.CDRInputStream
+import com.example.zenohapp.cdr.CDROutputStream
+import com.example.zenohapp.databinding.FragmentTeleopBinding
+import com.example.zenohapp.ros.AudioNote
+import com.example.zenohapp.ros.AudioNoteVector
+import com.example.zenohapp.ros.Battery
+import com.example.zenohapp.ros.Header
+import com.example.zenohapp.ros.Image
+import com.example.zenohapp.ros.Time
+import com.example.zenohapp.ros.Twist
+import com.example.zenohapp.ros.Vector3
+import io.zenoh.keyexpr.intoKeyExpr
+import io.zenoh.prelude.Encoding
+import io.zenoh.prelude.KnownEncoding
+import io.zenoh.publication.Publisher
+import io.zenoh.sample.Sample
+import io.zenoh.subscriber.Subscriber
+import io.zenoh.value.Value
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.cancelAndJoin
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import java.nio.ByteBuffer
+import kotlin.random.Random
+
+
+
+
+
+class TeleopFragment : Fragment(), OnTouchListener, OnClickListener, OnSeekBarChangeListener {
+    private var _binding: FragmentTeleopBinding? = null
+    private val binding get() = _binding!!
+
+
+    private lateinit var mFwdButton: ImageButton
+    private lateinit var mBwdButton: ImageButton
+    private lateinit var mLeftButton: ImageButton
+    private lateinit var mRightButton: ImageButton
+    private lateinit var mHomeButton: ImageButton
+    private lateinit var mSoundButton : ImageButton
+    private lateinit var mDeclareButton: Button
+    private lateinit var mNSEditText : EditText
+    private lateinit var mLinearSeekBar: SeekBar
+    private lateinit var mAngularSeekBar: SeekBar
+    private lateinit var mCameraView: ImageView
+    private lateinit var mBatteryBar : ProgressBar
+    private lateinit var viewModel: ZenohViewModel
+
+    private lateinit var mKeyExpr: String
+    private lateinit var mPublisher: Publisher
+    private lateinit var mBatterySubscriber: Subscriber<Channel<Sample>>
+    private lateinit var mTwist: Twist
+    private var isSending = false
+    private lateinit var mBatterySubJob : Job
+    private lateinit var mCameraSubJob : Job
+    private lateinit var mCommandPubJob : Job
+
+    private val actionDockSuffix = "/dock/_action/send_goal"
+    private val cmdVelSuffix = "/cmd_vel"
+    private val batteryStateSuffix = "/battery_state"
+    private val imageRawSuffix = "/oakd/rgb/preview/image_raw"
+    private var angularScale = 0.5
+    private var linearScale = 0.5
+    override fun onTouch(v: View?, event: MotionEvent?): Boolean {
```

Comment:
So whenever you touch the fragment layout, this callback is going to be triggered and it's going to check if the touch happened on a button?
I'd instead set for each button an implementation for [Button::onTouchEvent(MotionEvent event)](https://developer.android.com/reference/android/widget/TextView#onTouchEvent(android.view.MotionEvent)).